crates/datastore/src/locking_tx_datastore/committed_state.rs

```diff
@@ -345,19 +345,35 @@ impl CommittedState {
         Ok(())
     }
 
+    pub(super) fn replay_truncate(&mut self, table_id: TableId) -> Result<()> {
```

Comment:
`visit_delete` happens before `visit_truncate`, which removes rows from `st_*` tables, hence `schema_for_table` fails inside `visit_truncate`. I think we have to avoid calling that method.